include/nbl/system/ISkippableAsyncQueueDispatcher.h

```diff
@@ -0,0 +1,66 @@
+#ifndef __NBL_I_SKIPPABLE_ASYNC_QUEUE_DISPATCHER_H_INCLUDED__
+#define __NBL_I_SKIPPABLE_ASYNC_QUEUE_DISPATCHER_H_INCLUDED__
+
+#include "nbl/system/IAsyncQueueDispatcher.h"
+#include "nbl/system/SReadWriteSpinLock.h"
+
+namespace nbl {
+namespace system
+{
+
+struct SSkippableRequestBase : impl::IAsyncQueueDispatcherBase::request_base_t
+{
+    void set_skip_no_lock()
+    {
+        skip = true;
+    }
+    void set_skip()
+    {
+        write_lock_guard<> lk(rwlock);
+        set_skip_no_lock();
+    }
+    bool query_skip() const
+    {
+        bool val = false;
+        {
+            read_lock_guard<> lk(rwlock);
+            val = skip;
+        }
+        return val;
+    }
+
+    void lock()
+    {
+        rwlock.lock_write();
+    }
+    void unlock()
+    {
+        rwlock.unlock_write();
+    }
+
+    mutable SReadWriteSpinLock rwlock;
+
+private:
+    bool skip = false;
```

Comment:
`skip == future!=nullptr` always

so you could just replace the `skip` with an atomic future pointer.

and you probably dont need to lock the `rwlock` when querying a skip (`!future`), since its basically a binary semaphore